### ISSUE TITLE
Behavioural change for lighthouse: Use result: warning in certain cases (e.g., misconfiguration)

### ISSUE THREAD

**Issue by christian-kreuzberger-dtx** [keptn/keptn#1212]
At the moment (0.6.0.beta2) lighthouse will send an evaluation-done event in the following cases:

1. for performance tests: if the evaluation passes as it should
2. if no or just functional tests are executed 
3. for performance tests: if no slo.yaml is found 

See https://github.com/keptn/keptn/blob/fb73870afc87ad5c4465ee09ca3b8b8b8a217853/lighthouse-service/event_handler/start_evaluation_handler.go#L34-L83 for details

Depending on the use-case, it might be beneficial for 2. and 3. to cause a evaluation-done with `"result": "warning"` instead of `"result": "pass"`.

In addition, for some cases there is no evaluation-done event sent at all, e.g.:
https://github.com/keptn/keptn/blob/78ca2d4a6db087f3451d3565e13864a86e43a031/lighthouse-service/event_handler/start_evaluation_handler.go#L117-L122

Comments and opinions are welcome.

**Comment by johannes-b** [comment]
*Implemented feature:* In addressing this issue, I have implemented the functionality that an `evaluation-done` event with `result=failed` is sent when: 
- the environment variable `SLO_REQUIRED` is not set or it is set to `true`
- and neither an SLO nor SLI-provider is available for the evaluation.

If `SLO_REQUIRED=false` and no SLO is available, then the `evaluation-done` event with `result=warning` is sent.

--
Changing the lighthouse's behavior in regard to the functional test is quite problematic. I argue that we should not change it because a Keptn 0.6.1 user would get a warning in the "dev" stage when following our tutorials and best practices (our practice to execute functional tests in "dev"). In other words, just performance tests in "dev" can help a user to not get a "warning". 

I would ask the question, whether it is required to conduct an evaluation even though we already know that no performance data is available. An aspect that we will address with the new Shipyard format, which makes it explicit to conduct an evaluation.

**Comment by christian-kreuzberger-dtx** [comment]
At the moment, I believe that the Warning is appropriate and our best-practice is wrong (or we can not properly implement it with the current Keptn Version). 

For instance:

If an evaluation-done with "Passed" is sent, as a user, I would expect that something was evaluated. However, I, the user, have not defined such an evaluation. Therefore the warning is appropriate and should lead the user to one of the following actions:

* Disable evaluation for `dev` (currently not possible; will be possible with the new uniform/shipyard)
* Add an empty slo.yaml for dev

Just imagine a user that actually wants to execute **and evaluate** tests. This user will never be notified that the evaluation is not properly working.

So in short, regardless of the test-strategy (maybe except for *no strategy*), we should always try to execute the evaluation step and warn the user if something is missing (and maybe adapt our best practice).

**Comment by johannes-b** [comment]
The implementation lie described above has changed. 

* The environment variable `SLO_REQUIRED` has been removed
* The check which type of test has been executed has also been removed in order to not determine the behavior of the lighthouse-service depending on the executed test. 

Behavior: 
* If now SLO is found, an `evaluation-done` event with the result of the test task will be sent out. 
* If an SLO is available but no SLI-Provider is found, an `evaluation-done` with `result=failure` will be sent out.
